api/v1/bandwidth_test.go

```diff
@@ -0,0 +1,96 @@
+package api_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+
+	api "github.com/celestiaorg/bittwister/api/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func (s *APITestSuite) TestBandwidthStart() {
+	t := s.T()
+
+	reqBody := api.BandwidthStartRequest{
+		NetworkInterfaceName: s.ifaceName,
+		Limit:                100,
+	}
+	jsonBody, err := json.Marshal(reqBody)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest(http.MethodPost, "/api/v1/bandwidth/start", bytes.NewReader(jsonBody))
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	s.restAPI.BandwidthStart(rr, req)
+	// need to stop it to release the network interface for other tests
+	defer s.restAPI.BandwidthStop(rr, nil)
+
+	assert.Equal(t, http.StatusOK, rr.Code)
+}
+
+func (s *APITestSuite) TestBandwidthStop() {
```

Comment:
why not combine these tests? Separately out start and stop functionality just duplicates locs imo and combining them doesn't negatively impact the tests themselves.

Reply:
there is actually a case for including them, since the stop test doesn't use a `defer`, if they test failed it wouldn't release the network interface based on the comment in the start test. 

So they really should be combined and all the stop testing should be done in the defer statement to ensure that in the event of a failure we are always trying to stop the server.